vignettes/building_mocks.Rmd

```diff
@@ -0,0 +1,533 @@
+foundation of this function are the `label`, `column`, `param` and `values` 
+parameters. From here you can specify a `body_plan` where custom 
+`frmt_structures` can be added and built upon.
+
+**Note:** layering has been used to avoid repeating code; for 
+more information please see the "Reuse and layer tfrmts" article.
+
+
+## Making a simple mock table
+
+In the following example only one format structure has been specified, where 
+the desired outcome is:
+<ul>
+<li> the count value followed by the percent value </li>
+<li> the count value is formatted to be maximum three digits </li>
+<li> the percent value is formatted to be two digits rounded to one decimal place </li>
+</ul>
+
+```{r}
+mock_tbl <- function(tfrmt_obj){
+  tfrmt(
+    # Specify columns in the data
+    label = label,
+    column = column, 
+    param = param,
+    values = value,
+    sorting_cols = c(ord1),
+    
+    # Specify body plan
+    body_plan = body_plan(
+      frmt_structure(group_val = ".default", label_val = ".default",
+                     frmt_combine(
+                       "{count} {percent}",
+                       count = frmt("xxx"),
+                       percent = frmt_when("==100"~ frmt(""),
+                                           "==0"~ "",
+                                           "TRUE" ~ frmt("(xx.x%)"))))
+      ))
+}
+```
+
+The `print_mock_gt()` function has parameters `.default` and `n_cols` that 
+print three row labels and three columns by default - in this example there is 
+no need to supply data.
+
+You may enter different arguments into these parameters to alter the number of 
+rows and columns shown.
+
+```{r}
+# Print table
+mock_tbl() %>%
+  print_mock_gt()
+```
+
+If you would like to add a little more customisation by supplying your own mock 
+data you can utilise the `crossing()` function from the `{tidyr}` package. You 
+may also wish to add an integer column specifying order layer for complex tables.
+
+Here is an example:
+
+```{r}
+df <- crossing(label = c("label 1", "label 2", "label 3"),
+               column = c("trt1", "trt2", "trt1&trt2", "pl"),
+               param = c("count", "percent")) %>%
+  # Assign numerical order (optional)
+  mutate(ord1 = rep(seq(1:length(unique(.$label))), each = nrow(.)/length(unique(.$label)) ))
+
+df
+```
+
+This time the `df` data has been added to the `print_mock_gt()` function, so 
+now you can see that the label and column names have changed.
+
+```{r}
+# Print table
+mock_tbl() %>%
+  print_mock_gt(df)
+```
+
+You can adjust column names and structure using the `col_plan` parameter. It 
+accepts selection helpers from the `{tidyselect}` package. For example, you can 
+select `everything()` and remove columns that `starts_with("ord")`.
+
+You can also use `span_structure()` to add spanning labels as below.
+
+Note: You might see a warning along the lines of 
+`restarting interrupted promise evaluation`.
+
+```{r warning=FALSE}
+
+mock_tbl_adjust_cols <- function(){
```

Comment:
again, i'd probably avoid the use of the function and just pipe everything together

Reply:
see above